client.go

```diff
@@ -196,30 +186,37 @@ func (c *ClientConfig) verify() error {
 // server. If localAddr is empty client will always try to proxy to a local
 // port.
 func NewClient(cfg *ClientConfig) (*Client, error) {
+	if err := cfg.verify(); err != nil {
+		return nil, err
+	}
+
 	yamuxConfig := yamux.DefaultConfig()
 	if cfg.YamuxConfig != nil {
 		yamuxConfig = cfg.YamuxConfig
 	}
 
+	var proxy Proxy = &ProxyOverwrite{}
```

Comment:
@rjeczalik this allocation is on client creation not per request basis, per request code is later in proxy.go and it's like you said a defaultXXXProxy.